src/proxy/http/profiles.rs

```diff
@@ -477,23 +479,35 @@ pub mod router {
     where
         G: Stream<Item = Routes, Error = Never>,
         T: WithRoute + Clone,
-        T::Output: Eq + Hash,
+        T::Output: Clone + Eq + Hash,
         R: rt::Make<T::Output> + Clone,
         R::Value: svc::Service<http::Request<B>> + Clone,
     {
         fn update_routes(&mut self, routes: Routes) {
-            let slots = routes.len() + 1;
-            self.router = Router::new(
+            let default_route = self.target.clone().with_route(self.default_route.clone());
+
+            let capacity = routes.len() + 1;
+            let mut make = IndexMap::with_capacity(capacity);
+            make.insert(default_route.clone(), self.stack.make(&default_route));
+
+            for (_, route) in routes.clone().into_iter() {
```

Comment:
Not a big deal, but I would probably change this to
```rust
for (_, route) in &routes {
    let route = self.target.clone().with_route(route.clone());
    // ...
}
```
What you have currently is cloning the entire `IndexMap`, implying a heap allocation (and cloning the map's keys as well as the values, although the cloned keys are never used). On the other hand, iterating by reference and cloning each `route` will clone just the `Route`, which AFAIK is trivial to clone.

I dunno how hot this function is, so it may not be a big deal.